platform/android/java/lib/src/org/godotengine/godot/input/GodotInputHandler.java

```diff
@@ -156,6 +162,54 @@ public void run() {
 		return true;
 	}
 
+	public boolean onTouchEvent(final MotionEvent event) {
+		// Mouse drag (mouse pressed and move) doesn't fire onGeneticMotionEvent so this is needed
+		if (event.isFromSource(InputDevice.SOURCE_MOUSE)) {
+			if (event.getAction() != MotionEvent.ACTION_MOVE) {
+				// we return true because every time a mouse event is fired, the event is already handled
+				// in onGenericMotionEvent, so by touch event we can say that the event is also handled
+				return true;
+			}
+			return handleMouseEvent(event);
+		}
+
+		final int evcount = event.getPointerCount();
+		if (evcount == 0)
+			return true;
+
+		if (mRenderView != null) {
+			final float[] arr = new float[event.getPointerCount() * 3];
+
+			for (int i = 0; i < event.getPointerCount(); i++) {
+				arr[i * 3 + 0] = event.getPointerId(i);
+				arr[i * 3 + 1] = event.getX(i);
+				arr[i * 3 + 2] = event.getY(i);
+			}
+			final int pointer_idx = event.getPointerId(event.getActionIndex());
```

Comment:
Retrieve the `pointer_idx` in the `Runnable` below since it's only needed for certain action events.